linera-chain/src/certificate.rs

```diff
@@ -69,13 +75,25 @@ impl<T: Debug> Debug for GenericCertificate<T> {
     }
 }
 
-// NOTE: For backwards compatiblity reasons we serialize the new `Certificate` type as the old
-// one. Otherwise we would be breaking the RPC API schemas. We can't implement generic serialization
-// for `Certificate<T>` since only specific `T`s have corresponding `CertificateValue` variants.
 impl Serialize for ValidatedBlockCertificate {
```

Comment:
Don't we need to `impl Serialize for ConfirmedBlockCertificate`, too?

And maybe we should add a test that verifies that a `ConfirmedBlockCertificate` has a different hash than the equivalent `ValidatedBlockCertificate`?

Reply:
We don't have `ConfirmedBlockCertificate` as, at least not yet, it doesn't get ever (de)serialized. The other two (Timeout and ValidatedBlock) are part of the `ChainInfoManager` which enforces its fields to be deserializable.